mapstructure_test.go

```diff
@@ -2732,6 +2732,51 @@ func TestDecoder_IgnoreUntaggedFields(t *testing.T) {
 	}
 }
 
+func TestDecoder_Decode_SquashWithPrefix(t *testing.T) {
+	type Git struct {
+		Remote string `mapstructure:"remote"`
+	}
+
+	type GitHub struct {
+		Git   `mapstructure:"git,squash"`
```

Comment:
It seems not possible to use another name than `git` here. For example if you rename this to
```suggestion
		Git   `mapstructure:"gitx,squash"`
```
and update the `input` below accordingly (to `GITHUB_GITX_REMOTE`), the test panics.

Is this the desired behavior? I would have expected to be able to use any name here instead of just the name of the embedded struct.